src/firmware/app/primitives/direct_control_primitive.c

```diff
@@ -0,0 +1,147 @@
+            app_wheel_applyForce(app_firmware_robot_getBackRightWheel(robot),
+                                 control_msg.back_right_wheel_rpm);
+            app_wheel_applyForce(app_firmware_robot_getFrontRightWheel(robot),
+                                 control_msg.front_right_wheel_rpm);
+            break;
+        }
+        case DirectControlPrimitiveMsg_direct_velocity_control_tag:
+        {
+            DirectControlPrimitiveMsg_DirectVelocityControlMsg control_msg =
+                prim_msg.wheel_control.direct_velocity_control;
+            state->direct_velocity          = true;
+            state->direct_target_velocity_x = control_msg.velocity.x_component_meters;
+            state->direct_target_velocity_y = control_msg.velocity.y_component_meters;
+            state->direct_target_velocity_angular =
+                control_msg.angular_velocity.radians_per_second;
+            break;
+        }
+        default:
+        {
+            // Do nothing
+            state->direct_velocity = false;
+            assert(false);
+        }
+    }
+
+    switch (prim_msg.charge_mode)
+    {
+        case DirectControlPrimitiveMsg_ChargeMode_CHARGE:
+        {
+            app_charger_charge_capacitor(charger);
+            break;
+        }
+        case DirectControlPrimitiveMsg_ChargeMode_FLOAT:
+        {
+            app_charger_float_capacitor(charger);
+            break;
+        }
+        case DirectControlPrimitiveMsg_ChargeMode_DISCHARGE:
+        {
+            app_charger_float_capacitor(charger);
+            app_chicker_disableAutochip(chicker);
+            app_chicker_disableAutokick(chicker);
+            break;
+        }
+        default:
+        {
+            // discharge
+            app_charger_float_capacitor(charger);
+            app_chicker_disableAutochip(chicker);
+            app_chicker_disableAutokick(chicker);
+        }
+    }
+
+    switch (prim_msg.which_chick_command)
+    {
+        case DirectControlPrimitiveMsg_kick_speed_meters_per_second_tag:
+        {
+            app_chicker_kick(chicker,
+                             prim_msg.chick_command.kick_speed_meters_per_second);
+            break;
+        }
+        case DirectControlPrimitiveMsg_chip_distance_meters_tag:
+        {
+            app_chicker_chip(chicker, prim_msg.chick_command.chip_distance_meters);
+            break;
+        }
+        case DirectControlPrimitiveMsg_autokick_speed_meters_per_second_tag:
+        {
+            app_chicker_enableAutokick(
+                chicker, prim_msg.chick_command.autokick_speed_meters_per_second);
+            break;
+        }
+        case DirectControlPrimitiveMsg_autochip_distance_meters_tag:
+        {
+            app_chicker_enableAutochip(chicker,
+                                       prim_msg.chick_command.autochip_distance_meters);
+            break;
+        }
+        default:
+        {
+            // Do nothing
+            assert(false);
```

Comment:
Do we want to abort or to actually just do nothing?

Reply:
will log error in https://github.com/UBC-Thunderbots/Software/issues/1652 and not assert